projects/js-packages/components/components/jetpack-footer/index.tsx

```diff
@@ -1,12 +1,15 @@
-import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { __, _x } from '@wordpress/i18n';
 import { Icon, external } from '@wordpress/icons';
 import classnames from 'classnames';
 import React from 'react';
+import { getRedirectUrl } from '../..';
+import { STORE_ID as CONNECTION_STORE_ID } from '../../../../js-packages/connection/state/store';
```

Comment:
I wonder if this may prove problematic in the version of [the package that we ship to npm](https://www.npmjs.com/package/@automattic/jetpack-components).

Couldn't we bring in that data as a new component prop, to keep the component a bit simpler?

> Hardcoding the store id seemed error-prone.

That'd be my next suggestion, with a comment in the original file mentioning that it's copied there, but I'm not a regular component maintainer so I'll let others weigh in on this. :)

Reply:
Now that the connection package js is loaded externally and not bundled into any plugin after #38877, this import here can be problematic and result in duplicate store registration.

We should fix this in a better way. A few options we have:

- Let the consumer component pass the needed connection related information
- Make the component composable, which means the consumer component passes in the final output without this component here knowing anything about the connection data

Reply:
Another solution would be to use the data from the scrip-data package as explained in [this thread](https://github.com/Automattic/jetpack/pull/31627#discussion_r1251026389).

Reply:
That said, composability is still better because that will not make the component dependent upon anything related to jetpack connection or its logic.